File: ios/NORMALIZER/NORMALIZER/Models/Message.swift
import Foundation

struct ChatMessage: Codable, Identifiable, Sendable {
    let id: UUID
    let roomId: UUID
    let senderId: UUID
    let senderName: String
    let originalText: String
    let translatedText: String?
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case roomId = "room_id"
        case senderId = "sender_id"
        case senderName = "sender_name"
        case originalText = "original_text"
        case translatedText = "translated_text"
        case createdAt = "created_at"
    }
}

// WebSocket incoming message types
enum WSIncoming: Sendable {
    case message(ChatMessage)
    case userJoined(name: String)
    case userLeft(userId: String)
    case error(String)
}

File: backend/app/schemas/message.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class MessageRead(BaseModel):
    id: uuid.UUID
    room_id: uuid.UUID
    sender_id: uuid.UUID
    sender_name: str
    original_text: str
    translated_text: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}
